src/plugins/intel_gpu/src/graph/eltwise.cpp

```diff
@@ -107,6 +107,140 @@ layout eltwise_inst::calc_output_layout(eltwise_node const& node, kernel_impl_pa
+                                                       eltwise_mode::max,
+                                                       eltwise_mode::mod,
+                                                       eltwise_mode::eq,
+                                                       eltwise_mode::ne,
+                                                       eltwise_mode::lt,
+                                                       eltwise_mode::le,
+                                                       eltwise_mode::gt,
+                                                       eltwise_mode::ge,
+                                                       eltwise_mode::squared_diff,
+                                                       eltwise_mode::floor_mod,
+                                                       eltwise_mode::logic_and,
+                                                       eltwise_mode::logic_or,
+                                                       eltwise_mode::logic_xor};
+
+        OPENVINO_ASSERT((std::find(eltwise_int_modes.begin(), eltwise_int_modes.end(), mode) != eltwise_int_modes.end()),
+                            desc->id + "Requested eltwise mode is not supported for integer types.");
+    }
+
+    // Logic and comparison operations should return i8 for any inputs
+    std::vector<eltwise_mode> eltwise_bool_modes = {eltwise_mode::eq,
+                                                    eltwise_mode::ne,
+                                                    eltwise_mode::lt,
+                                                    eltwise_mode::le,
+                                                    eltwise_mode::gt,
+                                                    eltwise_mode::ge,
+                                                    eltwise_mode::logic_and,
+                                                    eltwise_mode::logic_or,
+                                                    eltwise_mode::logic_xor};
+    if (std::find(eltwise_bool_modes.begin(), eltwise_bool_modes.end(), mode) != eltwise_bool_modes.end()) {
+        output_layout.data_type = data_types::i8;
+    }
+
+    output_layout.data_type = desc->output_data_type.value_or(output_layout.data_type);
+
+    if (impl_param.has_fused_primitives()) {
+        output_layout.data_type = impl_param.get_fused_output_layout().data_type;
+    }
+
+    if (!desc->stride.empty()) {
+        auto new_size = input_layout.get_tensor();
+        // we can safely use only first stride, since we're using first input, and input / stride should give exact same
+        // value for every input
+        new_size.spatial[0] = (input_layout.spatial(0) - 1) / desc->stride[0].spatial[0] + 1;
+        new_size.spatial[1] = (input_layout.spatial(1) - 1) / desc->stride[0].spatial[1] + 1;
+        new_size.spatial[2] = (input_layout.spatial(2) - 1) / desc->stride[0].spatial[2] + 1;
+        input_layout.set_tensor(new_size);
```

Comment:
This part should also be adjusted for PartialShape based flow.

Reply:
Fixed it